src/providers/cambridge/utils.py

```diff
@@ -0,0 +1,31 @@
+import requests
+
+def get_page_content(url):
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an HTTPError if the response status code indicates an error
+        return response.text
+    except requests.exceptions.HTTPError as err:
+        print(f"HTTP Error: Failed to fetch the webpage ({err})")
```

Comment:
You can also change fprints to structlog. Here you can find step by step example: https://www.structlog.org/en/stable/bound-loggers.html#step-by-step-example

From docs, you can see that there are different log methods like: debug(), info(), warning(), error(), and critical(). In this case we need error().